Lib/tempfile.py

```diff
@@ -775,8 +775,8 @@ def writelines(self, iterable):
 
 
 class TemporaryDirectory(object):
-    """Create and return a temporary directory.  This has the same
-    behavior as mkdtemp but can be used as a context manager.  For
+    """Create a temporary directory, returning its name. This has the
+    same behavior as mkdtemp but can be used as a context manager.  For
     example:
```

Comment:
This returns the path only when it is used as a context manager [1]. When you create a temporary directory using `tempfile.TemporaryDirectory()`, you get an object of `TemporaryDirectory`, not the name as this change states. So this is inaccurate. 

```
In [13]: t = tempfile.TemporaryDirectory()

In [14]: t
Out[14]: <TemporaryDirectory '/var/folders/wr/lsnd74td3yb5q2nkxhxxnr980000gn/T/tmp1utsvx08'>

In [15]: type(t)
Out[15]: tempfile.TemporaryDirectory

In [16]: with tempfile.TemporaryDirectory() as td:
    ...:     print(td)
    ...:
/var/folders/wr/lsnd74td3yb5q2nkxhxxnr980000gn/T/tmp5d_ye7qi
```

[1] https://github.com/python/cpython/blob/master/Lib/tempfile.py#L803-L804

Reply:
Suggestion:

```python
"""Context manager to create temporary directories.

This has the same behavior as mkdtemp but is meant to be used as
a context manager.  For example:

    with TemporaryDirectory() as tmpdir:
        ...

Entering the context returns a string containing the full path
of the directory; exiting the context removes the directory and
everything contained in it.
"""
```